src/TitleDetails/TitleDetails.js

```diff
@@ -1,30 +1,107 @@
-import React from 'react';
+import React, { useCallback } from 'react';
 import PropTypes from 'prop-types';
-import { get } from 'lodash';
+import {
+  get,
+  sortBy,
+} from 'lodash';
 import { FormattedMessage } from 'react-intl';
 
 import {
   Accordion,
   AccordionSet,
+  Button,
   Col,
+  ConfirmationModal,
   ExpandAllButton,
   Pane,
-  Row,
   PaneMenu,
-  Button,
+  Row,
 } from '@folio/stripes/components';
 import { IfPermission } from '@folio/stripes/core';
 import { ViewMetaData } from '@folio/stripes/smart-components';
-import { useAccordionToggle } from '@folio/stripes-acq-components';
+import {
+  useAccordionToggle,
+  useModalToggle,
+} from '@folio/stripes-acq-components';
 
 import TitleInformation from './TitleInformation';
+import ExpectedPiecesList from './ExpectedPiecesList';
+import ReceivedPiecesList from './ReceivedPiecesList';
+import AddPieceModal from './AddPieceModal';
 import {
-  TITLE_ACCORDION,
+  ORDER_FORMAT_TO_PIECE_FORMAT,
+  ORDER_FORMAT,
+  PIECE_FORMAT_OPTIONS,
+  PIECE_STATUS,
   TITLE_ACCORDION_LABELS,
+  TITLE_ACCORDION,
 } from './constants';
+import { getPOLLocationsForSelect } from './utils';
 
-const TitleDetails = ({ onClose, title, poLine, onEdit }) => {
+const TitleDetails = ({
+  locations,
+  onAddPiece,
+  onCheckIn,
+  onClose,
+  onEdit,
+  pieces,
+  poLine,
+  title,
+}) => {
   const [expandAll, sections, toggleSection] = useAccordionToggle();
+  const [isAcknowledgeNote, toggleAcknowledgeNote] = useModalToggle();
+  const [isAddPieceModalOpened, toggleAddPieceModalOpened] = useModalToggle();
+  const receivingNote = get(poLine, 'details.receivingNote');
+  const expectedPieces = pieces.filter(({ receivingStatus }) => receivingStatus === PIECE_STATUS.expected);
+  const receivedPieces = sortBy(pieces.filter(({ receivingStatus }) => receivingStatus === PIECE_STATUS.received), 'receivedDate');
```

Comment:
too long line, please break it down, in the meantime I'll try to fix linter for such cases

Reply:
done